int64 m_XCentroid
int64 m_YCentroid
bool m_reset